File: chainOfResponsability/pyhton/soporte/soporte_primer_nivel.py
from soporte.soporte import Soporte

class SoportePrimerNivel(Soporte):
    def manejar_solicitud(self, solicitud):
        if "reinicio" in solicitud:
            return "Soporte Básico: Realizando reinicio de sistema"

        if self.proximo_nivel:
            return self.proximo_nivel.manejar_solicitud(solicitud)

        return "Ningún nivel de soporte puede manejar la solicitud"

File: chainOfResponsability/php/src/SoportePrimerNivel.php
<?php
namespace SoporteTecnico;
require_once('Soporte.php');

use SoporteTecnico\Soporte;

class SoportePrimerNivel implements Soporte {
    private ?Soporte $proximoNivel = null;

    public function setProximoNivel(Soporte $siguiente): void {
        $this->proximoNivel = $siguiente;
    }

    public function manejarSolicitud(string $solicitud): ?string {
        if (strpos($solicitud, 'reinicio') !== false) {
            return "Soporte Básico: Realizando reinicio de sistema";
        }

        if ($this->proximoNivel !== null) {
            return $this->proximoNivel->manejarSolicitud($solicitud);
        }

        return "Ningún nivel de soporte puede manejar la solicitud";
    }
}
?>